src/RadixTree.elm

```diff
@@ -110,209 +70,16 @@ singleton element =
-                    treeZipper
-
-            else
-                -- We have a partial match with the existing tree. We need to add a new
-                -- parent tree.
-                let
-                    rightTreeUpdated =
-                        treeZipper
-                            -- Keep only the diff elements of the Right element.
-                            |> TreeZipper.replaceLabel splitInfo.right
-                            -- Get the new tree from the focus point.
-                            |> TreeZipper.tree
-
-                    newTree =
-                        Tree.singleton splitInfo.intersect
-                            |> Tree.prependChild rightTreeUpdated
-                            |> Tree.prependChild (Tree.singleton splitInfo.left)
-                in
-                treeZipper
-                    |> TreeZipper.replaceTree newTree
-
-
-{-| Return a record holding the matching elements, the ones the differ from the "left"
-(i.e. new items being inserted), and the ones on the "right" (i.e. out of an existing
-node of the tree).
--}
-split : ( List a -> List a -> List a, List a -> List a -> List a ) -> List a -> List a -> { intersect : List a, left : List a, right : List a }
-split ( intersectFunc, removeFunc ) xs ys =
-    let
-        intersect =
-            intersectFunc xs ys
-
-        left =
-            removeFunc xs intersect
-
-        right =
-            List.foldl (\x accum -> List.Extra.remove x accum) ys intersect
-    in
-    { intersect = intersect
-    , left = left
-    , right = right
-    }
-
-
-{-| Find the matching elements, regardless of their order in the list.
--}
-unOrderedIntersect : List a -> List a -> List a
-unOrderedIntersect xs ys =
-    List.foldr
-        (\x accum ->
-            if List.member x ys then
-                x :: accum
-
-            else
-                accum
-        )
-        []
-        xs
-
-
-{-| Remove matching elements, with no regard to of their order in the list. That is,
-even if the element is in the middle or end of the list, it will be removed.
--}
-unOrderedRemove : List a -> List a -> List a
-unOrderedRemove xs intersect =
-    List.foldl (\x accum -> List.Extra.remove x accum) xs intersect
-
-
-{-| Find the matching elements, with regard to their order in the list. That is,
-we look for matches starting from the first element.
--}
-orderedIntersect : List a -> List a -> List a
-orderedIntersect xs ys =
-    let
-        listLength =
-            List.length xs
-    in
-    -- We start with entire list, and trim it down to find the intersecting elements.
-    if List.Extra.isPrefixOf xs ys then
-        xs
-
-    else if listLength == 1 then
-        -- No match
-        []
-
-    else
-        let
-            xsUpdated =
-                List.take (listLength - 1) xs
-        in
-        orderedIntersect xsUpdated ys
-
+        |> RadixTree.toTree
+        |> Tree.children --> [Tree.singleton [4], Tree.singleton [3]]
```

Comment:
I find it a bit hard to understand what to expect when using `Tree.children`. Maybe a string-representation in a comment could give a better indication? (And then a manually written test to not lose the test)